### ISSUE TITLE
Credo complains when there's pattern match with `=` in a lambda inside of an `if` condition

### ISSUE THREAD

**Issue by ashton314** [rrrene/credo#841]
### Environment

* Credo version (`mix credo -v`): 1.5.1
* Erlang/Elixir version (`elixir -v`): Elixir 1.11.0 (compiled with Erlang/OTP 23)
* Operating system: macOS 10.15.7

### What were you trying to do?

```elixir
if Enum.any?(Keyword.get_values(changeset.errors, attr), &({^message, _} = &1)) do
  ...
end
```
Note how there is a pattern match in the predicate to `Enum.any?`.

### Expected outcome

No issues from Credo

### Actual outcome

Credo complains with `[F] → There should be no matches in 'if' conditions.`, even though this is not an assignment. My guess is it gets upset that it sees a lone `=` in the conditional expression of an `if`.

It might make sense to to a more nuanced check to see if the top-level expression of the conditional is a pattern-matching assignment.

**Comment by rrrene** [comment]
The check works as intended. We can, of course, always discuss whether or not the check has the right intention ;-)

From the explanation (`mix credo explain Credo.Check.Refactor.MatchInCondition`):

```
      While this fine:

          # okay, simple wildcard assignment:

          if contents = File.read!("foo.txt") do
            do_something()
          end

      the following should be avoided, since it mixes a pattern match with a
      condition and do/else blocks.

          # considered too "complex":

          if {:ok, contents} = File.read("foo.txt") do
            do_something()
          end
```

This check intends to have only trivial conditionals in `if` macros (note the quotes in the last comment, where a tagged tuple match is described as "complex" in the context of this check).

As mentioned, we can always discuss whether or not this is a valid intention, but I am sure you wouldn't argue that 

```
Enum.any?(Keyword.get_values(changeset.errors, attr), &({^message, _} = &1))
```

is a more complex pattern match than

```
{:ok, contents} = File.read("foo.txt")
```

(the one described as "complex" in the check's description)

**Comment by ashton314** [comment]
I don't argue that my match expression is any less complex than the expression given in the example, but I think this should be allowed for a different reason than the pure complexity of a match.

Credo doesn't complain if I have a function like this:

```elixir
pred = &({^message, _} = &1)
```

and it doesn't complain if I *use* that predicate like this in an `if` conditional:

```elixir
if Enum.any?(Keyword.get_values(changeset.errors, attr), pred), do: ...
```

Furthermore, credo doesn't find any issues with the complex match inside a lambda as an argument to a function:

```elixir
_foo = Enum.any?(Keyword.get_values(changeset.errors, attr), &({^message, _} = &1))
```

So I think that, by transitivity, that expression should be permissible in the conditional of an `if`. Since that pattern match is inside of a function, I'd want that function to be opaque to credo in this instance. (I.e. Credo would check the anonymous function without considering the surrounding context of the `if` macro.)

I understand the warning when we get an expression like this:

```elixir
if {:ok, foo} = some_func(), do: ...
```

because *the pattern match is the highest-level expression in the AST of the conditional*, and that would be a better use for a `case` statement. However, in the case where there happens to be a match inside *an anonymous function*, and one that is *not* at the root level in the conditional AST, then I don't think credo should treat that in the same way.

**Comment by rrrene** [comment]
As I said, whether or not a check is valid for one's own use-case is debatable by design.

The one thing that gets me about this is the fact that I don't see a possible parameter to make this customizable. As in: I don't get how this should work (quite possibly due a lack of understanding, imagination and/or sleep).

What I mean is this: If we were debating the `if {:ok, foo} = some_func(), do: ...` case, we could agree on a check parameter called `:allow_tagged_tuples`  that gives people a choice.

What would such a parameter look like in your case?

**Comment by rrrene** [comment]
This is part of v1.6.1 💚
